packages/browser/tests/metrics/persistentSessionProviderTests.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { TimeHelper } from '@backtrace/sdk-core/lib/common/TimeHelper';
+import { BacktraceBrowserSessionProvider } from '../../src/BacktraceBrowserSessionProvider';
+describe('Session provider tests', () => {
+    it('Should generate a new uuid on new session', () => {
+        const sessionProvider = new BacktraceBrowserSessionProvider();
+
+        expect(sessionProvider.sessionId).toBeDefined();
+    });
+
+    it('Should reuse the same sessionId', () => {
+        const sessionProvider1 = new BacktraceBrowserSessionProvider();
+        const sessionProvider2 = new BacktraceBrowserSessionProvider();
+        expect(sessionProvider1.sessionId).toEqual(sessionProvider2.sessionId);
+    });
+
+    it('Should generate a new sessionId if the lastActive timestamp is greater than persistence interval time', () => {
+        const fakeId = 'test';
+        const lastSessionActiveDate = new Date(Date.now() - BacktraceBrowserSessionProvider.PERSISTENCE_INTERVAL - 1);
+        localStorage.setItem('backtrace-last-active', lastSessionActiveDate.getTime().toString(10));
+        localStorage.setItem('backtrace-guid', fakeId);
+
+        const sessionProvider = new BacktraceBrowserSessionProvider();
+        expect(sessionProvider.sessionId).toBeDefined();
+        expect(sessionProvider.sessionId).not.toEqual(fakeId);
+    });
```

Comment:
I would also do an "anti-test" of that. So something like "Should not generate a new sessionId if the lastActive timestamp is less than persistence interval time"